homework/index.js

```diff
@@ -30,13 +30,103 @@
     return elem;
   }
 
+  function removeChildren(element) {
+    while (element.firstChild) {
+      element.removeChild(element.firstChild);
+    }
+  }
+
   function main(url) {
     fetchJSON(url, (err, data) => {
       const root = document.getElementById('root');
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        const jsonString = JSON.stringify(data, null, 2);
+        const menu = createAndAppend('div', root, {
+          id: 'menuSelect',
+        });
+        createAndAppend('h2', menu, { text: 'HYF Repositories' });
+        const select = createAndAppend('select', menu, {
+          id: 'selection',
+        });
+        const repoDetails = createAndAppend('div', root, {
+          id: 'repoDetails',
+        });
+        createAndAppend('p', repoDetails, {
+          id: 'repoDescriptionP',
+        });
+        createAndAppend('p', repoDetails, {
+          id: 'forks',
+        });
+
+        createAndAppend('p', repoDetails, {
+          id: 'updateDate',
+        });
+
+        const renewalSpace = createAndAppend('div', root, {
+          id: 'contributors',
+        });
+
+        let valueNumber = 0;
+        const repositories = JSON.parse(jsonString, null);
+        let contributorsURL = '';
+
+        repositories.forEach(repo => {
+          const options = createAndAppend('option', select, {
+            value: valueNumber,
+          });
+          options.textContent = repo.name;
+          valueNumber += 1;
+        });
+        const selectElement = document.getElementById('selection');
```

Comment:
There is no need to get a reference to the `select` element like this as you already have it from line 50.

Reply:
Done!